src/nav/ModalNavBar.js

```diff
@@ -1,33 +1,74 @@
 import React from 'react';
-import { View, Text } from 'react-native';
+import { View, Text, StyleSheet } from 'react-native';
 import { connect } from 'react-redux';
+import Icon from 'react-native-vector-icons/Entypo';
 
 import boundActions from '../boundActions';
 import { CONTROL_SIZE } from '../common/platform';
-import { styles } from '../common';
 import { BRAND_COLOR } from '../common/styles';
+import { styles, Touchable, SearchInput } from '../common';
 import NavButton from './NavButton';
 
+
+const navBarStyles = StyleSheet.create({
+  touchable: {
+    paddingLeft: 10,
+    paddingRight: 10,
+  },
+});
+
+
 class ModalNavBar extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      isSearchEnabled: false
+    };
+  }
 
   props: {
     nav: any,
     title: string,
-    popRoute: () => void,
-  }
+    searchBarOnChange: () => {}
+  };
+
+  state: {
+    isSearchEnabled: boolean
+  };
 
   render() {
-    const { nav, title, popRoute } = this.props;
+    const { isSearchEnabled } = this.state;
+    const { nav, title, popRoute, searchBarOnChange } = this.props;
     const textStyle = [styles.navTitle, nav.index > 0 && { marginRight: CONTROL_SIZE }];
 
+    const onMagnifyingGlassPress = () => {
```

Comment:
not sure how I could make instance method from it.
When I pass it to `SearchInput` component `this` was changed and state inside wasn't visible.